### PR TITLE
Time travel cleanup

### PR DESCRIPTION
**Pull request by Frawstcrabs**
Adds a debounce to the time travel range input, so that you can scroll through the timeline a bit without having it immediately jump as soon as you click. The block number shown and the date of the current block also update in real time as you scroll.

### CODE REVIEW

File: dapp/src/components/TimeTravelWidget.tsx [hunk 1]
```diff
@@ -93,20 +95,43 @@ export default function TimeTravelWidget({
...
   );
 
+  const debouncedBlockChange = useMemo(
+    () => _.debounce(handleBlockChange, 300),
+    [handleBlockChange]
+  );
+
+  const onBlockChange = useCallback(
+    (block: string) => {
+      setCurrentBlock(block);
```
Comment (opatdchan) [inline dapp/src/components/TimeTravelWidget.tsx:105]: Thank you for the PR! 
However, you stumbled upon a problem I found no way around, which is caused by this line: when `currentBlock` changes, the `bbnQuery` fires on every single change, which means if you slide around a a bit you can potentially make thousands of queries, which is great for query fees but really not ideal, whereas if you only leave the debounced block, the UI lags.
I don't have much experience with React Hooks, so I wouldn't know how to work around it.

**Review by opatdchan — CHANGES_REQUESTED** [review]

File: dapp/src/components/TimeTravelWidget.tsx (hunk 1 above)
Comment (Frawstcrabs) [reply dapp/src/components/TimeTravelWidget.tsx:105]: I think a solution might be to separate the block number being displayed from the block number that's actually used in the query, and only update the query number when `handleBlockChange` is run.
Sidenote, is there a way to easily tell when the client is sending a query?

Comment (Frawstcrabs) [reply dapp/src/components/TimeTravelWidget.tsx:105]: Wait, nevermind, I think I see the issue, the date doesn't update until the query is actually made.
Unless there's some kind of way to write a graph query which could basically let you map block numbers to dates using number ranges, I don't think there's much way around it.

**Comment by Frawstcrabs** [comment]
Turns out I completely forgot to update the number section of the widget when the date is changed. Now they should always be kept in sync whenever one of them is modified. The only downside is that the slider will now jump to the real block number after the debounce instead of staying put, but at least you can consider it to be more "correct" (plus I can't be bothered to hack around it).
I also made it so the widget manually calls queries instead of using `useQuery`, so the slider can ensure it only queries once the debounce is complete instead of querying every frame.

I don't think there's a way around the date not getting updated until the new block is actually travelled to (unless we somehow create an internal map of block number to date), but since the date is updated as soon as that happens, I honestly don't find it to be much of an issue.

**Pull request closed without merging**